libcodechecker/database_handler.py

```diff
@@ -147,9 +147,11 @@ def create_engine(connection_string):
 
         if make_url(connection_string).drivername == 'sqlite+pysqlite':
             # FIXME: workaround for locking errors
-            return sqlalchemy.create_engine(connection_string,
-                                            encoding='utf8',
-                                            connect_args={'timeout': 600})
+            return sqlalchemy.create_engine(
+                connection_string,
+                encoding='utf8',
+                connect_args={'timeout': 600,
+                              'check_same_thread': False})
```

Comment:
It is checked whether the database connection is used from only one thread in case SQLite, since it doesn't support multi-threaded usage. It results even an error message and stops the program. To be precise according to the documentations it is allowed to use SQLite in multi-threaded environment but it is the user's responsibility to avoid concurrent usage. In CodeChecker we meet this expectation so the error message can be silenced by this flag.